src/Kernel/client/kernel.ts

```diff
@@ -31,20 +31,43 @@ class Kernel {
                 console.log("my message received", message);
 
                 //create buttons as DOM objects in order to attach unique event handlers
-                let state_div = message.content.state.div_id;
+                let state: DisplayableState = message.content.state;
```

Comment:
If `state` is not going to be changed, I would use `const` instead. Same for the other variables. This helps other devs after you know the intent that the variable should not change!